### DISCUSSION TITLE
Easier way to create child spans

### DISCUSSION THREAD

**Discussion by aabmass** [open-telemetry/opentelemetry-js-api#29] [General]
Are there any plans to simplify the  `context.with()` situation to make it easy to create child spans? Right now, I believe the correct way to do this is:

```ts
const currentSpan = tracer.startSpan('outer');
context.with(api.setSpan(context.active(), currentSpan), () => {
  currentSpan.setAttribute(...);
  // ...
  someOtherTraceableCode(); // spans within here should become children of currentSpan
});
```

In python we have [`tracer.start_as_current_span()`](https://github.com/open-telemetry/opentelemetry-python/blob/4027dae7b7418b02630c82ecfc2cfe154f6560d2/opentelemetry-api/src/opentelemetry/trace/__init__.py#L286)) for this common case. I see this question come up a good amount on slack and I think this is probably what most users want when writing instrumentation.

**Comment by Flarna** [comment]
There is already a PR https://github.com/open-telemetry/opentelemetry-js-api/pull/14 and an issue where this was discussed https://github.com/open-telemetry/opentelemetry-js/issues/1923.

**Reply by aabmass** [reply]
Ahh thank you, ~I'll lock this and check that issue~ I actually can't, if someone can close this, that would be great

**Reply by dyladan** [reply]
Discussions are different than issues. There is no "open" or "closed." The best thing would be to mark the answer as the accepted answer.

Conversation locking is more of a moderation mechanism. If more information comes up later I want people to be able to comment here if needed.
